tests/aws-cpp-sdk-cognitoidentityprovider-integration-tests/OperationTest.cpp

```diff
@@ -0,0 +1,379 @@
+#include <aws/testing/TestingEnvironment.h>
+#include <aws/core/platform/Environment.h>
+#include <openssl/sha.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include <openssl/bn.h>
+#include <aws/core/utils/logging/LogMacros.h>
+
+
+using namespace Aws::CognitoIdentityProvider;
+using namespace Aws::CognitoIdentityProvider::Model;
+using namespace Aws::Client;
+using namespace Aws::Region;
+
+namespace
+{
+static const char* ALLOCATION_TAG = "IdentityProviderOperationTest";
+
+
+class IdentityProviderOperationTest : public ::testing::Test
+{
+public:
+    IdentityProviderOperationTest() :
+        client(nullptr)
+    {}
+
+    std::shared_ptr<Aws::CognitoIdentityProvider::CognitoIdentityProviderClient> client;
+    Aws::String testTrace;
+
+protected:
+
+    const Aws::String m_clientName{"testClient"};
+    Aws::String m_userPoolId;
+    Aws::String m_clientId;
+    const Aws::String m_username{"testUser"};
+    Aws::String m_clientSecret;
+
+    void SetUp()
+    {
+        Aws::Client::ClientConfiguration config;
+        config.region = AWS_TEST_REGION;
+
+        client = Aws::MakeShared<Aws::CognitoIdentityProvider::CognitoIdentityProviderClient>(ALLOCATION_TAG, config);
+
+        //make user pool
+        m_userPoolId = createUserPool("TestPool");
+
+        ASSERT_TRUE(!m_userPoolId.empty());
+
+        //make client
+        if(!m_userPoolId.empty())
+        {
+            auto ret = createPoolClient(m_userPoolId, m_clientName);
+            m_clientId = ret.first;
+            m_clientSecret = ret.second;
+        }
+
+        ASSERT_TRUE(!m_clientId.empty());
+        ASSERT_TRUE(!m_clientSecret.empty());
+
+        //create user
+        if(!m_userPoolId.empty() && !m_clientId.empty())
+        {
+            ASSERT_TRUE(createUser(m_username, "Password123!", m_userPoolId));
+        }
+
+    }
+
+    void TearDown()
+    {
+
+        //delete user
+        if(!m_userPoolId.empty())
+        {
+            ASSERT_TRUE(deleteUser(m_username, m_userPoolId));
+        }
+
+        // delete client
+        if(!m_userPoolId.empty() && !m_clientId.empty())
+        {
+            ASSERT_TRUE(deletePoolClient(m_userPoolId, m_clientId));
+        }
+
+        // delete user pool
+        if(!m_userPoolId.empty())
+        {
+            ASSERT_TRUE(deleteUserPool(m_userPoolId));
+        }
+
+
+        if (::testing::Test::HasFailure())
+        {
+            AWS_LOGSTREAM_DEBUG(ALLOCATION_TAG,  "Test traces: " << testTrace);
+        }
+        testTrace.erase();
+    }
+
+    private:
+    Aws::String createUserPool(Aws::String poolName)
+    {
+        if(poolName.empty())
+        {
+            return {};
+        }
+        Aws::CognitoIdentityProvider::Model::CreateUserPoolRequest userPoolRequest;
+        userPoolRequest.SetPoolName(poolName);
+
+        CreateUserPoolOutcome userPoolOutcome = client->CreateUserPool(userPoolRequest);
+        if (!userPoolOutcome.IsSuccess()) {
+            return {};
```

Comment:
this is a test so we should expecting/asserting service outcomes with with either [`AWS_ASSERT_SUCCESS` or `AWS_EXPECT_SUCCESS`](https://github.com/aws/aws-sdk-cpp/blob/b068f68fcb46e9bd4af397d3fa1e4f88f8e94b34/tests/testing-resources/include/aws/testing/AwsTestHelpers.h#L17-L27)